UI/ImDebugger/ImMemView.cpp

```diff
@@ -319,9 +319,43 @@ void ImMemView::ProcessKeyboardShortcuts(bool focused) {
 		if (ImGui::IsKeyPressed(ImGuiKey_PageUp)) {
 			ScrollWindow(-visibleRows_, GotoModeFromModifiers(false));
 		}
+
+		if(editableMemory_){
+			if(!asciiSelected_){
+				for(int i = ImGuiKey_0; i!=ImGuiKey_G; i++){
+					if(ImGui::IsKeyPressed(static_cast<ImGuiKey>(i))){
+						EditMemory(i-ImGuiKey_0);
+						return;
+					}
+				}
+				for(int i = ImGuiKey_Keypad0; i!=ImGuiKey_KeypadDecimal; i++){
+					if(ImGui::IsKeyPressed(static_cast<ImGuiKey>(i))){
+						EditMemory(i-ImGuiKey_Keypad0);
+						return;
+					}
+				}
+			}
+		}
 	}
 }
 
+void ImMemView::EditMemory(int newNibble){
+	uint8_t og= Memory::ReadUnchecked_U8(curAddress_);
+	u8 out;
+	if(selectedNibble_){
+		out = og ^((og^newNibble)&0xf);
```

Comment:
wouldn't this be simpler as just `(og & 0xF0) | newNibble` ? 

similarly below, `(og & 0x0F) | (newNibble << 4)`